src/wallet/coinselection.h

```diff
@@ -66,7 +71,10 @@ struct COutput {
     /** The fee required to spend this output at the consolidation feerate. */
     CAmount long_term_fee{0};
 
-    COutput(const COutPoint& outpoint, const CTxOut& txout, int depth, int input_bytes, bool spendable, bool solvable, bool safe, int64_t time, bool from_me, const std::optional<CFeeRate> feerate = std::nullopt)
+    /** The fee necessary to bump this UTXO's ancestor transactions to the target feerate */
+    CAmount ancestor_bump_fees{0};
+
+    COutput(const COutPoint& outpoint, const CTxOut& txout, int depth, int input_bytes, bool spendable, bool solvable, bool safe, int64_t time, bool from_me, const std::optional<CFeeRate> feerate = std::nullopt, const std::optional<std::reference_wrapper<interfaces::Chain>> chain_interface = std::nullopt)
```

Comment:
I have 2 concerns with this approach, please let me know what you think:

(1) The `COutput` constructor really shouldn't need to have a reference the chain interface; it smells a bit weird that coinselection (which I interpret to be a relatively well-modularized component thus far since it doesn't even depend on wallet) has a new dependency on interfaces/chain.h. It's unclear to me why `CalculateBumpFees()` needs to be called *inside* the constructor instead of just having `CAmount ancestor_bump_fees` be a parameter?

(2) This means `CalculateBumpFees()` will be called over and over again for each `COutput` constructed within the `AvailableCoins` loop. There's not a lot of duplicated work if all the outputs are from independent transactions, but it would definitely be faster to call it once with all the outpoints at once.

Approach-wise, I think it makes more sense to call `CalculateBumpFees()` just once, with the full list of outpoints. It should be fairly straightforward with preset inputs since you have the list already. For `AvailableCoins`, since they're constructed as you iterate through `mapWallet`, you could populate each output's `ancestor_bump_fees` values at the end. If you want to construct them as-is and then not mutate afterwards, then maybe do 2 passes for filtering and constructing?

Reply:
Now applying the bumpfees to each output after constructing the output.